src/main/java/de/tum/cit/aet/artemis/programming/service/ProgrammingMessagingService.java

```diff
@@ -196,22 +193,15 @@ public void notifyUserAboutNewResult(Result result, ProgrammingExerciseParticipa
      * @param studentParticipation the student participation for which Iris should be informed about the submission status
      */
     private void notifyIrisAboutSubmissionStatus(Result result, ProgrammingExerciseStudentParticipation studentParticipation) {
-        if (studentParticipation.getParticipant() instanceof User user) {
-            pyrisEventApi.ifPresent(eventApi -> {
-                final var exercise = studentParticipation.getExercise();
-                if (user.hasAcceptedExternalLLMUsage() && !exercise.isExamExercise()
-                        && irisSettingsApi.map(api -> api.isProgrammingExerciseChatEnabled(exercise.getId())).orElse(false)) {
-                    // Inform event service about the new result
-                    try {
-                        // This is done asynchronously to prevent blocking the current thread
-                        eventApi.trigger(new NewResultEvent(result));
-                    }
-                    catch (Exception e) {
-                        log.error("Could not trigger service for result {}", result.getId(), e);
-                    }
-                }
-            });
-        }
+        pyrisEventApi.ifPresent(eventApi -> {
+            // Inform event service about the new result
+            try {
+                eventApi.trigger(new NewResultEvent(result));
```

Comment:
can we pass the studentParticipation and/or exercise here to avoid loading them again later on?